src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/utils.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+// See: https://spdx.org/licenses/
+
+#ifndef UTILS_H_
+#define UTILS_H_
+
+#include <memory>
+#include <chrono>  // NOLINT
+#include <thread>  // NOLINT
+
+#define MAX_ARRAY_DIMS (5)
+#define SLEEP_US (1)
+
+namespace message_infrastructure {
+
+enum ProcessType {
+  ErrorProcess = 0,
+  ParentProcess = 1,
+  ChildProcess = 2
+};
+
+enum ChannelType {
+  SHMEMCHANNEL = 0,
+  RPCCHANNEL = 1,
+  DDSCHANNEL = 2
+};
+
+struct MetaData {
+  int64_t nd;
+  int64_t type;
+  int64_t elsize;
+  int64_t total_size;
+  int64_t dims[MAX_ARRAY_DIMS] = {0};
+  int64_t strides[MAX_ARRAY_DIMS] = {0};
+  void* mdata;
+};
+
+using MetaDataPtr = std::shared_ptr<MetaData>;
+
+namespace helper {
+
+static void Sleep() {
```

Comment:
For every use of this function, please provide a comment for why a std::promise<void/bool> (or other notification primitive) could not be used to immediately signal completion of whatever is being waited on?